packages/framework/src/Component/Elasticsearch/IndexDefinition.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Component\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\Exception\ElasticsearchIndexException;
+
+class IndexDefinition
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex
+     */
+    protected $index;
+
+    /**
+     * @var string
+     */
+    protected $definitionsDirectory;
+
+    /**
+     * @var string
+     */
+    protected $indexPrefix;
+
+    /**
+     * @var int
+     */
+    protected $domainId;
+
+    /**
+     * DocumentDefinition constructor.
+     *
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex $index
+     * @param string $definitionsDirectory
+     * @param string $indexPrefix
+     * @param int $domainId
+     */
+    public function __construct(AbstractIndex $index, string $definitionsDirectory, string $indexPrefix, int $domainId)
+    {
+        $this->index = $index;
+        $this->definitionsDirectory = $definitionsDirectory;
+        $this->indexPrefix = $indexPrefix;
+        $this->domainId = $domainId;
+    }
+
+    /**
+     * @return array
+     */
+    public function getDefinition(): array
+    {
+        $decodedDefinition = json_decode($this->getDefinitionFileContent(), true);
+        if ($decodedDefinition === null) {
+            throw ElasticsearchIndexException::invalidJsonInDefinitionFile(
+                $this->getIndex()->getName(),
+                $this->getDefinitionFilename()
+            );
+        }
+
+        return $decodedDefinition;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDefinitionFilename(): string
+    {
+        return $this->getDefinitionsDirectory() . $this->getIndex()->getName() . '/' . $this->getDomainId() . '.json';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDefinitionFileContent(): string
+    {
+        $definitionFilename = $this->getDefinitionFilename();
+        if (!is_readable($definitionFilename)) {
+            throw ElasticsearchIndexException::cantReadDefinitionFile($definitionFilename);
+        }
+
+        return file_get_contents($definitionFilename);
+    }
+
+    /**
+     * @return string
+     */
+    public function getDocumentDefinitionVersion(): string
```

Comment:
Should this method be public? Is it desirable to use it outside the class?

Reply:
Fixed. It should not be used elsewhere.